src/framework/core/js/RemoteModel.js

```diff
@@ -0,0 +1,237 @@
+/*
+Copyright 2017 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+var fluid_3_0_0 = fluid_3_0_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    fluid.defaults("fluid.remoteModelComponent", {
+        gradeNames: ["fluid.modelComponent"],
+        events: {
+            afterFetch: null,
```

Comment:
Having separate events for errors is ok, but I think separate onWrite/afterWrite and onFetch/afterFetch errors is dubious, because of the standard problem of associating together the particular firings of related events - that is, there is no clear way for a user to identify the *particular* firing of an afterFetch event that corresponds to a particular onFetch. For this reason I think we should combine these events and just have a more elaborate namespace structure within the fewer events that allows users to schedule things at a time of interest.

Reply:
@amb26 could you expand on what you mean by "have a more elaborate namespace structure within the fewer events that allows users to schedule things at a time of interest"

Reply:
I spoke with @amb26 in the IRC channel about this ( https://botbot.me/freenode/fluid-work/2018-02-26/?msg=97241487&page=1 ). The suggestion is to try to move the items from the afterWrite and afterFetch to be part of the onWrite and onFetch sequence. However, the current implementation is working, so will timebox the exploration of adjusting the implementation.

Reply:
I did some exploration into moving the afterFetch and afterWrite workflows into onFetch and onWrite respectively. The main issue preventing this is that the fireEventSequences used in the remote model, passes along the same payload to each listener ( https://github.com/jobara/infusion/blob/6c4381917a85641beb6195a50006976f7ebad733/src/framework/core/js/RemoteModel.js#L108-L126 ). This was done so that the listeners could be treated more like real listeners and not have to return back the payload at each step to the next listener in the chain. However, the onWrite and onFetch events don't have a payload. The payload was passed along in the afterWrite and afterFetch events.